### PR TITLE
added support for index pattern as param to CreateIndexMappings

### LINKED ISSUE #154: [FEATURE] CreateIndexMappings API - add support for index pattern
**Is your feature request related to a problem?**
CreateIndexMappings API now supports only concrete index as index_name parameter.

**What solution would you like?**
We need to be able to pass index pattern as index_name parameter. 

For example: 
```
{
    "index_name": "my_index*",
    "rule_topic": "netflow"
}
```

fixes #217

### PR DESCRIPTION
**Pull request by petardz**
Signed-off-by: Petar Dzepina <petar.dzepina@gmail.com>

### Description
added support for index pattern as param to CreateIndexMappings
 
### Issues Resolved
#154
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sbcd90** [comment]
hi @petardz , the pr looks good. can you resolve the `warning` messages also in the code if possible?

**Comment by sbcd90** [comment]
@petardz lets also discuss on enhancing this with `get mappings/view` api

**Comment by getsaurabh02** [comment]
@petardz Can we address the requested changes

**Review by sbcd90 — DISMISSED** [review]

File: src/main/java/org/opensearch/securityanalytics/transport/TransportCreateIndexMappingsAction.java [hunk 1]
```diff
@@ -43,11 +43,6 @@ public TransportCreateIndexMappingsAction(
     protected void doExecute(Task task, CreateIndexMappingsRequest request, ActionListener<AcknowledgedResponse> actionListener) {
         this.threadPool.getThreadContext().stashContext();
 
-        IndexMetadata index = clusterService.state().metadata().index(request.getIndexName());
-        if (index == null) {
-            actionListener.onFailure(new IllegalStateException("Could not find index [" + request.getIndexName() + "]"));
-            return;
-        }
```
[the remark this answers is not in the document]
Comment (petardz) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportCreateIndexMappingsAction.java:50]: No need I think. GET _mapping API will return mappings for all resolved indices in response.

**Comment by getsaurabh02** [comment]
@petardz Can we resolve the conflict. LGTM

**Pull request closed without merging**